docs/tutorial/truncating.md

```diff
@@ -0,0 +1,281 @@
+# Truncating
+
+Table lookups have a strict constraint on the number of bits they support. This can be limiting, especially if you don't need exact precision. As well as this, using larger bit-widths leads to slower table lookups.
+
+To overcome these issues, truncated table lookups are introduced. This operation provides a way to zero the least significant bits of a large integer and then apply the table lookup on the resulting (smaller) value.
+
+Imagine you have a 5-bit value, but you want to have a 3-bit table lookup. You can call `fhe.truncate_bit_pattern(input, lsbs_to_remove=2)` and use the 3-bit value you receive as input to the table lookup.
```

Comment:
Since the value is still 5-bits, this is not clear what happens here. Where does the 5 to 3bits conversion happens ?
The value is tagged as 3-bits compatible ? Or is it really a pattern truncated + TLU that is detected ?
What if the user puts a 5bits TLU ? 4bits TLU ?

Reply:
Ok so I would rephrase
"You can call `fhe.truncate_bit_pattern(input, lsbs_to_remove=2)` and use the 3-bit value you receive as input to the table lookup." by  something like that "A truncate enable a immediately following 5bits TLU to be optimized out (or converted) to an equivalent and faster 3bits table lookup.
 
For me what is important to say is, the it is an optimization/faster only when a TLU is after, and that the user table is still a 5bit. Otherwise it's not clear where the 3bits notion appears.

Reply:
Improved.